### ISSUE TITLE
Diffusion MLP initialization and training problem

### ISSUE THREAD

**Issue by Kevin-thu** [LTH14/mar#46]
Congrats for your great work! Got invaluable insights from this project.
I tried using diffusion loss to train a large autoregressive image generation transformer, but encountered some problems in the training of Diffusion MLP. I first wanted to overfit a small subset of a dataset (with ~100 training samples, batch size=1, learning rate=1e-4), but found that the diffusion loss just did not drop and converge even after 500k+ steps, and the generated samples remained noises. 
I dug deeply into the training process and found that the gradients wrt the Diffusion MLP module params are almost all zero (gradient vanishing) except the final layer, which implied that the MLPs haven't learn anything. 
I read about the initialization code of the MLP, and learned that if my understanding is right, the learning process of the Diffusion MLP is: the final layer -> the modulation layers -> the mlp layers. However, in my experiments, the last two parts (the modulation layers and the mlp layers) did not get any gradients throughout the training process. I also tried removing the zero initialization (just using the default initialization method), but found the diffusion loss exploded (and also gradient explosion). 
Could you provide insights into the possible reasons for these issues? Besides, do you have any suggestions for ensuring training stability? Your assistance would be greatly appreciated. Thanks!
<img width="509" alt="diffloss" src="https://github.com/user-attachments/assets/faf1106a-b7a2-4803-a3f9-b9e1946f2508">
![image](https://github.com/user-attachments/assets/d5c17d8f-6a68-4b3b-8eec-e45d29292cc8)

**Comment by LTH14** [comment]
This is weird -- in our case the diffloss drops quickly. Maybe you could consider a larger batch size? For example, a 128 batch size and only 1 iteration per epoch.
